packages/core/src/utils/promiseHandler.ts

```diff
@@ -0,0 +1,103 @@
+export interface EmbracePromiseRejectionConfig {
+    /**
+     * Enable/disable internal unhandled promise rejection handling.
+     */
+    enabled: boolean;
+
+    /**
+     * Log unhandled promise rejections to the console.
+     * Recommend to set to true in development environments for easier debugging.
+     * Recommend to set to false in production environments to avoid duplicate logging.
+     */
+    logToConsole: boolean;
+
+    /** 
+     * Optional custom handler for unhandled promise rejections.
+     * Use this to integrate with your own error tracking system.
+     * @example
+     * ```ts
+     * customHandler: (method, error) => {
+     *   MyErrorTrackingService.logError(`Unhandled promise rejection in ${method}`, error);
+     * }
+     * ```
+     */
+    customHandler?: (methodName: string, error: Error) => void;
+};
+
+let rejectionConfig: EmbracePromiseRejectionConfig = {
+    enabled: true,
+    logToConsole: typeof __DEV__ !== 'undefined' ? __DEV__ : false,
+};
+
+export function handleSDKPromiseRejection<T>(
+    promise: Promise<T>,
+    methodName: string,
+): void;
+
+export function handleSDKPromiseRejection(
+    methodName: string,
+    error: unknown,
+): void;
+
+export function handleSDKPromiseRejection<T>(
+    methodNameOrPromise: string | Promise<T>,
+    errorOrMethodName: unknown | string,
```

Comment:
seems like this method is only being invoked with a Promise and string, are the other types needed? Also seems like the generic isn't being used so maybe just:
```suggestion
export function handleSDKPromiseRejection(
    promise: Promise,
    methodName: string,
```